Sources/ProjectDescription/CompatibleXcodeVersions.swift

```diff
@@ -5,55 +5,76 @@ public enum CompatibleXcodeVersions: ExpressibleByArrayLiteral, ExpressibleByStr
     /// The project supports all Xcode versions.
     case all
 
+    case exact(Version)
+
+    case upToNextMajor(Version)
+
+    case upToNextMinor(Version)
+
     /// List of versions that are supported by the project.
-    case list([String])
+    case list([CompatibleXcodeVersions])
 
     // MARK: - ExpressibleByArrayLiteral
 
-    public init(arrayLiteral elements: [String]) {
-        self = .list(elements)
+    public init(arrayLiteral elements: [CompatibleXcodeVersions]) {
+        self = .list(elements.map { $0 })
     }
 
-    public init(arrayLiteral elements: String...) {
-        self = .list(elements)
+    public init(arrayLiteral elements: CompatibleXcodeVersions...) {
+        self = .list(elements.map { $0 })
     }
 
-    enum CodignKeys: String, CodingKey {
+    enum CodingKeys: String, CodingKey {
         case type
         case value
     }
 
     // MARK: - ExpressibleByStringInterpolation
 
     public init(stringLiteral value: String) {
-        self = .list([value])
+        self = .exact("\(value)")
     }
 
     // MARK: - Codable
 
     public func encode(to encoder: Encoder) throws {
```

Comment:
Don't we have automatic synthesize since 5.5? We can probably drop the whole custom Codable conformance

Reply:
This has the small side effect of slightly changing the output of `tuist dump`. Let me know if you want this listed as a breaking change.

Reply:
Maybe we can remove all the not required custom codable in 3.0? @fortmarek @pepicrft

Reply:
`tuist dump` output is not really meant to be stable, so I think it's fine to make it breaking now.